src/org/traccar/protocol/WatchProtocolEncoder.java

```diff
@@ -15,6 +15,8 @@
  */
 package org.traccar.protocol;
```

Comment:
This class probably shouldn't extend string encoder. It doesn't really return strings.

Reply:
Also thought, but it still uses formatCommand() of StringProtocolEncoder in formatTextCommand. If I don't extend StringProtocolEncoder then I'll have to find another way to do it, which would probably involve code duplication. I could make an inner class in WatchProtocolEncoder that extends StringProtocolEncoder and includes the formatValue. (Thoufh formatCommand is protected...) and we would also have another "copy" of BaseProtocolEncoder inside WatchProtocolEncoder.

Reply:
OK, we can probably keep it.